ui-admin/src/forms/FormDesigner.tsx

```diff
@@ -26,14 +28,40 @@ export const FormDesigner = (props: FormDesignerProps) => {
           onSelectElement={setSelectedElementPath}
         />
       </div>
-      <div className="flex-grow-1 overflow-scroll">
+      <div className="flex-grow-1 overflow-scroll py-2 px-3">
         {(() => {
           if (selectedElementPath === undefined) {
             return (
               <p className="mt-5 text-center">Select an element to edit</p>
             )
           }
 
+          const selectedElement = get(selectedElementPath, value) as FormContentPage | FormElement
+
+          if (!('type' in selectedElement) && !('questionTemplateName' in selectedElement)) {
```

Comment:
this logic feels like belongs in a helper function

Reply:
I put it here because this way TypeScript handles the type narrowing and provides some type safety around deciding which component to render based on the selected element. With a user defined type guard (ex: `const isPage = (element: FormElement | FormContentPage): element is FormContentPage => { ... }`, TS trusts the developer that whatever logic in the type guard is correct.